hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ProcessorState.java

```diff
@@ -30,9 +31,13 @@
     PROCESS_WATERMARK,
 
     /**
-     * Making calls to {@link Processor#tryProcess()} and {@link
-     * Processor#process(int, com.hazelcast.jet.core.Inbox)} until the inbox
-     * is empty.
+     * Making calls to {@link Processor#tryProcess()} until it returns true.
+     */
+    NULLARY_TRY_PROCESS,
```

Comment:
People who don't understand it should ignore it  :-)
We use this term for some time already to distinguish the nullary `tryProcess` from the non-nullary one in `AbstractProcessor`. The term is internal, so no worry about confusing the users.

Reply:
Well readability is subjective. We added "nullary" because without it we confused it for the non-nullary `tryProcess` in  `AbstractProcessor`.

The "try" part means the method returns boolean if it cannot finish now. I'd rather remove that part: `NULLARY_PROCESS`. It's the only "try" in `ProcessorState`.

Reply:
If i saw the name `TRY_PROCESS`, I'd automatically think it refers to the one taking parameters, because that is the "main" `tryProcess`, while the nullary one is needed only for special purposes that aren't data-driven. This name does use a less-frequent term, but it's still well in the "easily googlable" category (typing "nullary" gives me "nullary function" as the first suggestion).

I would also not like to see `NULLARY_PROCESS`, I'd think "process" is a noun here. So at least for me the current name works the best.